Makefile

```diff
@@ -1019,10 +1034,13 @@ LINT_CPP_FILES = $(filter-out $(LINT_CPP_EXCLUDE), $(wildcard \
 	tools/icu/*.h \
 	))
 
-lint-cpp:
+lint-cpp: tools/.cpplintstamp
+
+tools/.cpplintstamp: $(LINT_CPP_FILES)
 	@echo "Running C++ linter..."
 	@$(PYTHON) tools/cpplint.py $(LINT_CPP_FILES)
```

Comment:
Replace `$(LINT_CPP_FILES)` with `$?` and it will run the linter only for files that changed since last time.

It would be nice to do the same thing with tools/check-imports.py but that script will need a tweak to accept a list of file names.